crates/stages/src/stages/execution.rs

```diff
@@ -260,34 +270,34 @@ fn execution_checkpoint<DB: Database>(
         // Otherwise, we recalculate the whole stage checkpoint including the amount of gas
         // already processed, if there's any.
         _ => {
-            let processed = calculate_gas_used_from_headers(tx, 0..=start_block - 1)?;
+            let processed = calculate_gas_used_from_headers::<DB>(provider, 0..=start_block - 1)?;
```

Comment:
dont love how these become more verbose, would push it to be on the provider itself, e.g. on header provider